events/importer/enkora.py

```diff
@@ -1200,72 +1698,101 @@ def _ranges_overlap(
+            course["reservation_event_group_name"]
+        )
+        if self.LANGUAGE_SWEDISH in title_kw_ids:
             event_language = "sv"
-            desc_kw_ids.remove(self.LANGUAGE_SWEDISH)
-        if self.LANGUAGE_ENGLISH in desc_kw_ids:
+            title_kw_ids.remove(self.LANGUAGE_SWEDISH)
+        if self.LANGUAGE_ENGLISH in title_kw_ids:
             event_language = "en"
-            desc_kw_ids.remove(self.LANGUAGE_ENGLISH)
-        for kw_id in desc_kw_ids:
+            title_kw_ids.remove(self.LANGUAGE_ENGLISH)
+        for kw_id in title_kw_ids:
             if kw_id in self.AUDIENCES:
                 audience_kw_ids.add(kw_id)
             else:
                 sport_kw_ids.add(kw_id)
 
-        # post-process
-        audience_kws = []
-        for kw_id in audience_kw_ids:
-            kw = Keyword.objects.get(id=kw_id)
-            audience_kws.append(kw)
-        sport_kws = []
-        for kw_id in sport_kw_ids:
-            kw = Keyword.objects.get(id=kw_id)
-            sport_kws.append(kw)
+        # Determine shop link:
+        liikuntakauppa_fi_link = self.liikuntakauppa_link(sport_kw_ids)
 
-        return event_language, audience_kws, sport_kws, min_age, max_age
+        return (
+            event_language,
+            audience_kw_ids,
+            sport_kw_ids,
+            min_age,
+            max_age,
+            liikuntakauppa_fi_link,
+        )
 
     @staticmethod
-    def _parse_description_age(description: str) -> tuple[Optional[int], Optional[int]]:
+    def liikuntakauppa_link(keywords: set) -> str:
+        if not keywords:
+            # We didn't manage to match to any sport.
+            chosen_link = EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT
+        else:
+            sports_with_keyword = set(keywords).intersection(
+                EnkoraImporter.liikuntakauppa_sport.keys()
+            )
+            if (
+                sports_with_keyword
+                and len(sports_with_keyword) == 1
+                and EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT in sports_with_keyword
+            ):
+                # The only match is into default sport
+                chosen_link = EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT
+            else:
+                if EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT in sports_with_keyword:
+                    sports_with_keyword.remove(EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT)
+
+                if not sports_with_keyword:
+                    raise ValueError(
+                        "Enkora importer internal error: Keyword '{}' has no Liikuntakauppa definition! "
+                        "Refusing to continue.".format(", ".join(list(keywords)))
+                    )
+
+                # Return first one
+                chosen_keyword = next(iter(sports_with_keyword))
+                chosen_link = EnkoraImporter.liikuntakauppa_sport[chosen_keyword]
+
+        return EnkoraImporter.liikuntakauppa_links[chosen_link]
```

Comment:
Change: Reduce the cognitive complexity a bit:
```suggestion
    def liikuntakauppa_link(keywords: set) -> str:
        if not keywords:
            # We didn't manage to match to any sport.
            return EnkoraImporter.liikuntakauppa_links[EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT]

        sports_with_keyword = set(keywords).intersection(
            EnkoraImporter.liikuntakauppa_sport.keys()
        )
        if (
            sports_with_keyword
            and len(sports_with_keyword) == 1
            and EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT in sports_with_keyword
        ):
            # The only match is into default sport
            return EnkoraImporter.liikuntakauppa_links[EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT]

        if EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT in sports_with_keyword:
            sports_with_keyword.remove(EnkoraImporter.LIIKUNTAKAUPPA_DEFAULT)

        if not sports_with_keyword:
            raise ValueError(
                "Enkora importer internal error: Keyword '{}' has no Liikuntakauppa definition! "
                "Refusing to continue.".format(", ".join(list(keywords)))
            )

        # Return first one
        chosen_keyword = next(iter(sports_with_keyword))
        chosen_link = EnkoraImporter.liikuntakauppa_sport[chosen_keyword]

        return EnkoraImporter.liikuntakauppa_links[chosen_link]
```

Reply:
Approved change.